### PR TITLE
chore: 🤖 use lumberjack library builds for compatibility tests

### PR DESCRIPTION
**Pull request by NachoVazquez**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ x ] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ x ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

`Lumberjack` host application is being built and test using `Lumberjack` library sources instead of build bundles. This is a poor compatibility test since is not being tested in the same way it would be used by npm clients.

CI process includes a step before building and testing the app where the path to the library is modified to look at the dist folder.

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -108,11 +119,27 @@ jobs:
...
+        run: yarn use:dist
+
+      - name: Cache library build artifacts
+        uses: actions/cache@v2
+        id: lib-build-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
+        with:
+          path: dist/ngworker/lumberjack
+          key: ${{ runner.os }}-node-${{ matrix.node-version }}-angular-${{ matrix.angular-version }}-lumberjack-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-node-${{ matrix.node-version }}-angular-${{ matrix.angular-version }}-lumberjack
```
Comment (LayZeeDK) [inline .github/workflows/ci.yml:132]: I think we need a dash at the end of the restore key. At least that's the way it's done in other steps.

File: package.json [hunk 2]
```diff
@@ -21,7 +21,8 @@
     "release": "cd libs/ngworker/lumberjack && standard-version --infile ../../../CHANGELOG.md",
     "publish": "cd dist/ngworker/lumberjack && npm publish",
     "ci": "yarn install && yarn lint && yarn build:lib && yarn test:internal:ci && yarn test:lib:ci && yarn build && yarn test:ci && yarn e2e",
-    "format": "npx prettier --config prettier.config.js --write \"**/*.*\" \"!dist/**\" \"!yarn.lock\""
+    "format": "npx prettier --config prettier.config.js --write \"**/*.*\" \"!dist/**\" \"!yarn.lock\"",
+    "use:dist": "json -I -f tsconfig.json -e 'this.compilerOptions.paths[\"@ngworker/lumberjack\"] = [\"dist/ngworker/lumberjack\"]; this.compilerOptions.paths[\"@ngworker/lumberjack/*-driver\"] = [\"dist/ngworker/lumberjack/*-driver\"]'"
```
Comment (LayZeeDK) [inline package.json:25]: I think the driver path should be removed for this job. The packaged bundle should include these entry points as well so its path is enough.

Comment (NachoVazquez) [reply package.json:25]: That was my first thought, but it failed to find those modules.

That made me think that it is a problem we may face with the npm package as well.

We may need to add an index.ts in the drivers' folder.

Comment (NachoVazquez) [reply package.json:25]: I just tried, moving the **dist** to `node-modules` and it worked. We are ok with the npm version. 

It is weird, however, that it doesn't work using the path in tsconfig.

Comment (NachoVazquez) [reply package.json:25]: It work with this config `"@ngworker/*": ["dist/ngworker/*"],`

Comment (LayZeeDK) [reply package.json]: Let's go with that.

**Comment by LayZeeDK** [comment]
Okay, now we need to see if we can reproduce the broken pipeline steps locally to figure out what's going on.

**Review by LayZeeDK — APPROVED** [review]

**Pull request merged**